storage/storage.go

```diff
@@ -2420,6 +2420,7 @@ type retryConfig struct {
 	policy      RetryPolicy
 	shouldRetry func(err error) bool
 	maxAttempts *int
+	maxDuration time.Duration
```

Comment:
Also let's call it `maxRetryDuration` since it doesn't actually limit the total duration of calls.

Reply:
Done.
It stutters a bit (`retryConfig.maxRetryDuration`) but that was my initial instinct as well. It's clearer anyway since `retryConfig` can have any name.